### PR TITLE
Use new rapids-cython component of rapids-cmake to simplify builds

### PR DESCRIPTION
**Pull request by vyasr**
The custom CMake code written in python/cmake has now been generalized and ported to rapids-cmake so that it can be shared across rapids libraries.

This PR depends on rapidsai/rapids-cmake#184 and tests will not pass until that is merged.

### CODE REVIEW

**Comment by vyasr** [comment]
rerun tests

File: python/CMakeLists.txt [hunk 1]
```diff
@@ -16,9 +16,9 @@ cmake_minimum_required(VERSION 3.20.1 FATAL_ERROR)
 
 set(rmm_version 22.06.00)
 
-set(CYTHON_FLAGS
-    "--directive binding=True,embedsignature=True,always_allow_keywords=True"
-    CACHE STRING "The directives for Cython compilation.")
+file(DOWNLOAD https://raw.githubusercontent.com/rapidsai/rapids-cmake/branch-22.06/RAPIDS.cmake
```
Comment (bdice) [inline python/CMakeLists.txt:19]: This probably requires an update to https://github.com/rapidsai/rmm/blob/branch-22.06/ci/release/update-version.sh

File: python/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +51,7 @@ if(NOT RMM_FOUND)
   add_subdirectory(../ rmm-cpp)
 endif()
 
-add_subdirectory(cmake)
+include(rapids-cython)
+rapids_cython_init()
```
Comment (bdice) [inline python/CMakeLists.txt:55]: This seems like it stands alone. A newline between might be helpful.
```suggestion
rapids_cython_init()

```

**Review by bdice — COMMENTED** [review]
This needs a hook in the version updater script, but I think it's fine otherwise!

File: python/CMakeLists.txt (hunk 1 above)
Comment (vyasr) [reply python/CMakeLists.txt:19]: Good catch.

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

**Review by robertmaynard — APPROVED** [review]

**Review by jakirkham — APPROVED** [review]
Thanks Vyas and everyone for the reviews! 😄

**Comment by jakirkham** [comment]
@gpucibot merge

**Pull request merged**